### PR TITLE
Fix CCDownloader-curl will crash in js-test & Fix broken image link in TextureCacheTest.js

### PR DESCRIPTION
**Pull request by xiaofeng11**

### CODE REVIEW

File: tests/js-tests/src/TextureCacheTest/TextureCacheTest.js [hunk 1]
```diff
@@ -79,9 +79,9 @@ var TextureLoadImgTest = TextureCacheTestBase.extend({
         });
         this.addChild(this._labelSecond, 1);
 
-        var url = "http://cn.cocos2d-x.org/image/logo.png";
+        var url = "http://www.cocos2d-x.org/images/logo.png";
         cc.textureCache.addImageAsync(url, this.texFirstLoaded, this);
-        cc.textureCache.addImageAsync(url, this.texSecondLoaded, this);
```
Comment (pandamicro) [inline tests/js-tests/src/TextureCacheTest/TextureCacheTest.js:84]: Why remove the second one ?

Comment (xiaofeng11) [inline tests/js-tests/src/TextureCacheTest/TextureCacheTest.js:84]: Sorry, I comment the code for debug and forget to roll back

File: cocos/network/CCDownloader-curl.cpp [hunk 2]
```diff
@@ -845,7 +847,7 @@ namespace cocos2d { namespace network {
             DLLOG("    DownloaderCURL: finish Task: Id(%d)", coTask.serialId);
         }
         
-        if (_impl->stoped())
+        if (!_breakScheduleFunc && _impl->stoped())
```
Comment (pandamicro) [inline cocos/network/CCDownloader-curl.cpp]: When the current object became wild pointer, we can't assume its property value remains the same, so there may still be problems

Comment (xiaofeng11) [inline cocos/network/CCDownloader-curl.cpp]: Current object destruct is raised by a function inside this object, so I think before the function exist, the object won't be really released.

Comment (pandamicro) [inline cocos/network/CCDownloader-curl.cpp]: But its destructor have been called, are you sure ?

Comment (xiaofeng11) [inline cocos/network/CCDownloader-curl.cpp]: I use this way to process class with sub-thread inside it on pc before. It works.
On mobile platform need tester do more test on variable device to confirm .

**Comment by pandamicro** [comment]
Great !

**Pull request merged**